src/fireedge/src/modules/features/General/slice.js

```diff
@@ -0,0 +1,324 @@
+        state.selectedIds = payload
+      })
+      .addCase(actions.updateDisabledSteps, (state, { payload }) => {
+        state.disabledSteps = payload
+      })
+      .addCase(actions.setUpdateDialog, (state, { payload }) => {
+        state.isUpdateDialog = !!payload
+      })
+      .addCase(actions.setFullMode, (state, { payload }) => {
+        state.isFullMode = !!payload
+      })
+      .addCase(actions.setBreadcrumb, (state, { payload }) => {
+        state.breadcrumb = payload
+      })
+      .addCase(actions.setTableViewMode, (state, { payload }) => {
+        state.tableViewMode = payload
+      })
+      .addCase(actions.setLoadOsProfile, (state, { payload }) => {
+        const { stepId } = payload
+        const exists = get(state.loadedOsProfile, stepId, false)
+
+        if (!exists) {
+          set(state.loadedOsProfile, stepId, true)
+        }
+      })
+
+      .addCase(actions.resetLoadOsProfile, (state) => {
+        state.loadedOsProfile = {}
+      })
+
+      /* FIELD MODIFICATIONS */
+      .addCase(actions.setFieldPath, (state, { payload }) => {
+        state.fieldPath = payload
+      })
+      .addCase(actions.resetFieldPath, (state) => {
+        state.fieldPath = ''
+      })
+      .addCase(actions.initModifiedFields, (state, { payload }) => {
+        // Get field path and check if there is something in this path
+        const fieldPath = state?.fieldPath || ''
+        const exists = get(state.modifiedFields, fieldPath, false)
+
+        // If the path has content, do nothing
+        if (!exists) {
+          set(state.modifiedFields, fieldPath, payload)
+        }
+      })
+      .addCase(actions.changePositionModifiedFields, (state, { payload }) => {
+        // Get from payload the path to the arrays and the number of the element
+        const sourcePath = payload?.sourcePath
+        const sourcePosition = payload?.sourcePosition
+        const targetPath = payload?.targetPath
+        const sourceDelete = payload?.sourceDelete
+        const emptyObjectContent = payload?.emptyObjectContent
+        const targetPosition = payload?.targetPosition
+
+        // Get source element and change to target array and delete on source array
+        const sourceArray = get(state.modifiedFields, sourcePath)
+        const targetArray = get(state.modifiedFields, targetPath)
+
+        const sourceIndex = calculateIndex(sourceArray, sourcePosition)
+        const targetIndex = calculateIndex(targetArray, targetPosition)
+
+        const sourceFinalPosition =
+          sourceIndex === -1 ? sourceArray.length : sourceIndex
+        const targetFinalPosition =
+          targetIndex === -1 ? targetArray.length : targetIndex
+
+        targetArray[targetFinalPosition] = sourceArray[sourceFinalPosition]
+
+        sourceDelete && pullAt(sourceArray, sourceFinalPosition)
+        emptyObjectContent &&
+          (sourceArray[sourceFinalPosition] = pickBy(
+            sourceArray[sourceFinalPosition],
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).